### DISCUSSION TITLE
Workaround for working with another member

### DISCUSSION THREAD

**Discussion by Bong1hong** [Budibase/budibase#10160] [Help] [answered]
Hi,

My colleague and I are thinking of working simultaneously on a Budibase app.
I'm aware that the version control features are still in-development #4360

Just wonder is there any temporary workaround for co-development of a Budibase app.
(merging of exported app, etc....) 

Thanks.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Bong1hong,

This is something which is planned to be worked on. However, there is no exact timeline on when this will be done.

Currently there are no workarounds that I'm aware of.

I hope this helps!

**Comment by dewdeded** [comment]
We have a second test instance, so one guy can develope screens (in the live-system), the other can test (the current project quality or possible solutions for still existing problems).
Bigger automations or javascript code can be concurrently developed (and tested) if you have more than one Budibase instance. (We dont like "dev copies" of apps in development on the live system.) Our strategy is to only have (nearly) productive BB apps in the main instance. Sooner or later there will be enterprise feature and BB will have a dev, testing/staging and production enviroment in one installation. I am sure. (n8n for example is implementing such features currently.)

**Back to your original question:**
Their are alot of tasks that can be done simultaneously while developing a BB app with more than one dev. (Writing technical documentation, preparing data, lecture and inform end users, report to management, do work on the connected systems & services where the BB app is integrated, develope ideas for app improvements, react to user feedback, assign new app users to Budibase portal and app, create end user documentation, talk with Legal or Data privacy team about the currently developed app, ...)

All our BB apps have at least two developers assigned as responsible developer and are created together. One dev only working and knowing an app is super dangerous. (Because no one can support the app, when he is not avaible.) Any app that is important for your organisation must from the project start have at least two devs. (I normally also include more people for the app team, like people from IT support (so they can later help), end users as product owners and first testers, other IT admins that are responsible for the data/results/new processes that the new BB app produces, etc.)

If you want to work with more people on one BB app, its a good idea to assign roles and responsibilities for the project.
One guy is concentrating on screens/layout, the other on data and automations, someone else is focussing on support and project management, etc. So you have nearly zero problems with app locks. If an app is locked, where I need to work I call my co-worker and he either lifts the lock in < 1 min or he completes the task for me, because its done super quick.

So much text, but I just wanted to say: we work since 1,5 years with Budibase and we have have four active BB developers (one even full time) and app locks were never a problem for us.

**Reply by Bong1hong** [reply]
You are right with the dangerous of the only one developer (me previously). That's the main reason this question is asked as more developer is being assign to help offload my tasks.

Thanks for advices and suggestion especially on the project management and effective communication related topic ☺️
